### DISCUSSION TITLE
half-second stutter during gameplay (osu!)

### DISCUSSION THREAD

**Discussion by Mschweit** [ppy/osu#26349] [Q&A]
Summary:
A Stutter of about half a second occurs at 0:03 in the attached video and 1:24 in the attached replay.
logs, replay, and video taken showing the replay attached, as I was not able to reproduce.

Reason for opening Discussion:
Opening this discussion because I haven't seen a stutter this bad in lazer in probably more than six months and also saw this in the logs:
2024-01-03 00:48:51 [verbose]: Playback discrepancy detected (1 of allowed 5): 86,300.5 vs 87,330.5

Platform:
Windows 10

https://github.com/ppy/osu/assets/37157907/72460719-06d7-4b1d-b75f-fde126a6fa40

[compressed-logs.zip](https://github.com/ppy/osu/files/13815138/compressed-logs.zip)
[Toa of guts playing sekai - Sensei no Koto Suki ni Nacchau (Petal) [Chitose_s Another] (2024-01-02_19-50).zip](https://github.com/ppy/osu/files/13815141/Toa.of.guts.playing.sekai.-.Sensei.no.Koto.Suki.ni.Nacchau.Petal.Chitose_s.Another.2024-01-02_19-50.zip)

**Comment by Mschweit** [comment]
I had another large stutter with a similar log output on a different map:
2024-01-07 04:07:24 [verbose]: Playback discrepancy detected (1 of allowed 5): 126,825.4 vs 127,315.2

This session I had the nvidia options: Prefer Max Performance, and Threaded Optimization on, but a couple apps were running in the background.  I'll close them from now on and see if I still experience any large stutters.
![image](https://github.com/ppy/osu/assets/37157907/fe157388-36af-44ab-b1b7-341ebb91758c)
[compressed-logs_2024-01-06.zip](https://github.com/ppy/osu/files/13852246/compressed-logs_2024-01-06.zip)
7c8f4a4-db00-4290-8d73-6d221a0a3e7b)

**Comment by Mschweit** [comment]
Experienced another large stutter even though there were no (large) apps running in the background with the same type of log output:
2024-01-10 00:27:57 [verbose]: Playback discrepancy detected (1 of allowed 5): 37,186.6 vs 37,557.2

Since this doesn't seem to be an issue with my machine, I'll start recording my play sessions with frame statistics on to try to catch it.

System tray during play session:
![image](https://github.com/ppy/osu/assets/37157907/f51a6b26-a7cf-44c1-8e5a-080527674fb8)


[compressed-logs_2021-01-09.zip](https://github.com/ppy/osu/files/13881530/compressed-logs_2021-01-09.zip)

**Reply by peppy** [reply]
I've already found some issues with the amount of memory allocations we're performing under certain circumstances. Stay tuned for the next release where this should be heavily alleviated.
